server/src-lib/Hasura/GraphQL/Execute/RemoteJoins.hs

```diff
@@ -106,50 +133,18 @@ rebuildFieldStrippingRemoteRels =
                            , rrAlias = _fAlias subfield
                            , rrAliasIndex = idx
                            , rrPhantomFields =
-                               map
-                                 G.unName
-                                 (filter
-                                    (`notElem` fmap _fName (_fSelSet field0))
-                                    (hasuraFieldNames remoteField))
+                               coerce $ toList $
+                                 hasuraFieldNames `Set.difference` siblingSelections
                            }
-      let fields = rights selSetEithers
-      pure
-        field0
-          { _fSelSet =
-              Seq.fromList $
-                selSetEithers >>= \case
-                  Right field -> [field]
-                  Left remoteField ->
-                    mapMaybe
-                      (\name ->
-                         if elem name (map _fName fields)
-                           then Nothing
-                           else Just
-                                  (Field
-                                     { _fAlias = G.Alias name
-                                     , _fName = name
-                                     , _fType =
-                                         G.NamedType (G.Name "unknown3")
-                                     , _fArguments = mempty
-                                     , _fSelSet = mempty
-                                     , _fRemoteRel = Nothing
-                                     }))
-                      (hasuraFieldNames remoteField)
-          }
+                       hasuraFieldNames = 
+                         coerceSet $ rtrHasuraFields $ rmfRemoteRelationship remoteField
       where
         thisPath = parentPath <> RelFieldPath (pure (idx0, _fAlias field0))
-        hasuraFieldNames =
-          map (G.Name . getFieldNameTxt) . toList . rtrHasuraFields . rmfRemoteRelationship
+        siblingSelections = Set.fromList $ toList $ fmap _fName $ _fSelSet field0
 
--- | Get a list of fields needed from a hasura result.
-neededHasuraFields
-  :: RemoteField -> [FieldName]
-neededHasuraFields = toList . rtrHasuraFields . rmfRemoteRelationship
 
 -- | Join the data returned from the original hasura request with the remote values.
-joinResults :: GQRespValue
-            -> [(Batch, EncJSON)]
-            -> GQRespValue
+joinResults :: GQRespValue -> [(Batch, EncJSON)] -> GQRespValue
```

Comment:
Would have been more readable if the signature was:

`joinResults :: GQRespValue -> [(RemoteRelField, EncJSON)] -> GQRespValue`


See comment on `data Batch`